microprofile/messaging/core/src/test/java/io/helidon/microprofile/messaging/connector/ConnectorConfigTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2025 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.microprofile.messaging.connector;
+
+import io.helidon.common.reactive.Multi;
+import io.helidon.microprofile.tests.junit5.AddConfig;
+import org.eclipse.microprofile.config.Config;
+import io.helidon.config.mp.MpConfig;
+import io.helidon.microprofile.messaging.MessagingCdiExtension;
+import io.helidon.microprofile.tests.junit5.AddBean;
+import io.helidon.microprofile.tests.junit5.AddExtension;
+import io.helidon.microprofile.tests.junit5.DisableDiscovery;
+import io.helidon.microprofile.tests.junit5.HelidonTest;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import org.eclipse.microprofile.reactive.messaging.Channel;
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.messaging.spi.Connector;
+import org.eclipse.microprofile.reactive.messaging.spi.IncomingConnectorFactory;
+import org.eclipse.microprofile.reactive.streams.operators.PublisherBuilder;
+import org.eclipse.microprofile.reactive.streams.operators.ReactiveStreams;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@Disabled("because @Inject and @HelidonTest(resetPerTest=true) throws exception")
```

Comment:
This is a little hard  for me :) . Let me explain.

The same tests were added for 4.x and it works there, but for 3.x I have this exception
<img width="1693" alt="Screenshot 2025-03-12 at 09 46 34" src="https://github.com/user-attachments/assets/8bd74000-9c6f-4821-89c3-a65d2dca1a43" />

The error message is clear. I can't inject, because of this parameter `resetPerTest=true`. It works for 4.x after [this refactoring](https://github.com/helidon-io/helidon/commit/1baabd5042389661b75238c774c38aceb132bd6b#diff-5dea65dfdf039d5e5a05021730d568db7512b0734e5580b1cc2239432b766267L156) .

But how can I fix this test for 3.x? Or should I remove these tests at all, because `AdHocConfigBuilderTest` does all necessary checks?

Reply:
The test class only has one method so you don't need to use `resetPerTest = true`.
You can move the `@AddConfig` and `@AddBean` annotations on the class.

Reply:
Thank you